platform/darwin/src/MGLGeoJSONSource.mm

```diff
@@ -82,9 +93,8 @@ - (void)commonInit
         _features = nil;
     } else if (self.geoJSONData) {
         NSString *string = [[NSString alloc] initWithData:self.geoJSONData encoding:NSUTF8StringEncoding];
-        const auto geojson = mapbox::geojson::parse(string.UTF8String).get<mapbox::geojson::feature_collection>();
+        const auto geojson = mapbox::geojson::parse(string.UTF8String);
         source->setGeoJSON(geojson);
-        _features = MGLFeaturesFromMBGLFeatures(geojson);
```

Comment:
We need a `shape` property that can be any MGLShape (or any MGLFeature). We can have a `features` property as a convenience for when `shape` happens to contain an MGLShapeCollectionFeature.